### PR TITLE
Filthy feed posts [READY]

### PR DESCRIPTION
**Pull request by cybrox**
Allows users to mark feed posts as nsfw
Users with sfw-filter enabled won't see these posts.

Copy of #372

### CODE REVIEW

File: app/frontend/app/templates/story/comment.hbs [hunk 1]
```diff
@@ -52,6 +52,11 @@
 </div>
 
 <div class="story-actions">
+  {{#if adult}}
+    <span class="nswf-indicator" data-tooltip="This post has been marked as nsfw!">
```
Comment (NuckChorris) [inline app/frontend/app/templates/story/comment.hbs]: >nswf
>not safe work for

Comment (cybrox) [inline app/frontend/app/templates/story/comment.hbs]: Not safe wor fork!

**Comment by vikhyat** [comment]
why is the build #failing?

**Comment by cybrox** [comment]
I don't know, it just aborted the test without running into any errors.
I can't manually restart it though (afaik), you might want to do that.

File: app/frontend/app/components/post-comment.js [hunk 2]
```diff
@@ -32,9 +50,18 @@ export default Ember.Component.extend({
...
   actions: {
     submitPost: function() {
-      var comment = this.get('newPost');
-      this.set('newPost', "");
-      this.sendAction('action', comment);
+      var comment = this.get('newPost'),
+          isAdult = this.get('newPostAdult');
+
+      this.setProperties({
+        'newPost': "",
```
Comment (vikhyat) [inline app/frontend/app/components/post-comment.js]: Does the key need to be quoted here?

Comment (cybrox) [inline app/frontend/app/components/post-comment.js]: No, it'll work either way though.
Should I change it?

**Comment by vikhyat** [comment]
Looks good at first glance, will do a more thorough review tomorrow.

File: app/frontend/app/components/post-comment.js [hunk 3]
```diff
@@ -1,23 +1,41 @@
...
+        self.$(".panel-footer").slideDown(200);
+      } else {
         self.$(".status-form").val('');
         self.$(".panel-footer").slideUp(200, function() {
           self.$(".status-form").trigger("autosize.destroy");
         });
       }
-    });
-  },
+    }, 300);
```
Comment (NuckChorris) [inline app/frontend/app/components/post-comment.js]: What's up with the `setTimeout` here? Is it a workaround for a bug?  If so, you should probably add a comment explaining that

Comment (cybrox) [inline app/frontend/app/components/post-comment.js]: It's a delay for the form blur so you can technically click the nsfw button before entering any text. I'll add a comment and squash.

**Comment by NuckChorris** [comment]
Once that comment is shoved in, LGTM.  I don't see any potential blockers (though it might bring out some unnoticed bugs in the feed code)

**Pull request merged**